test/Dialect/FIRRTL/imconstprop.mlir

```diff
@@ -347,6 +347,35 @@ firrtl.circuit "invalidReg2"   {
 
 // -----
 
+// CHECK-LABEL: firrtl.circuit "RegResetInvalid"
+firrtl.circuit "RegResetInvalid"  {
+  firrtl.module @RegResetInvalid(in %clock: !firrtl.clock, in %a: !firrtl.uint<1>, out %b: !firrtl.uint<1>) {
+    %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    %r = firrtl.regreset %clock, %a, %c1_ui1  : !firrtl.uint<1>, !firrtl.uint<1>, !firrtl.uint<1>
+    %invalid_ui1 = firrtl.invalidvalue : !firrtl.uint<1>
+    firrtl.strictconnect %r, %invalid_ui1 : !firrtl.uint<1>
+    // CHECK: %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    // CHECK: firrtl.strictconnect %b, %c1_ui1 : !firrtl.uint<1>
+    firrtl.strictconnect %b, %r : !firrtl.uint<1>
+  }
+}
+
+// -----
+
+// CHECK-LABEL: firrtl.circuit "RegResetInvalidReset"
+firrtl.circuit "RegResetInvalidReset"  {
+  firrtl.module @RegResetInvalidReset(in %clock: !firrtl.clock, out %a: !firrtl.uint<1>) {
+    %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+    %invalid_ui1 = firrtl.invalidvalue : !firrtl.uint<1>
+    %r = firrtl.regreset %clock, %invalid_ui1, %c1_ui1  : !firrtl.uint<1>, !firrtl.uint<1>, !firrtl.uint<1>
+    firrtl.strictconnect %r, %invalid_ui1 : !firrtl.uint<1>
+    // CHECK: %invalid_ui1 = firrtl.invalidvalue : !firrtl.uint<1>
+    // CHECK: firrtl.strictconnect %a, %invalid_ui1 : !firrtl.uint<1>
+    firrtl.strictconnect %a, %r : !firrtl.uint<1>
```

Comment:
I _think_ this is okay, though I am somewhat nervous that this isn't getting IMCP'd/canonicalized immediately to constant zero. Any propagation of invalid is concerning and potentially a bug source.

This is not a blocking issue for this PR, though.

Reply:
I agree 100 percent, here is a follow up issue: https://github.com/llvm/circt/issues/2781

Reply:
First issue was bunk but maybe this second one is a real: https://github.com/llvm/circt/issues/2782